--- a/configure.ac
+++ b/configure.ac
@@ -94,7 +94,7 @@ AC_SUBST(GRADLE)
 dnl ================================================================
 dnl C99 compiler checks.
 dnl ================================================================
-
+AC_SUBST(HOST_BUILD_DIR)
 AC_LANG_PUSH(C)
 AC_MSG_CHECKING([if $CC supports C99 without any flags])
 AC_COMPILE_IFELSE(
--- a/Makefile.am
+++ b/Makefile.am
@@ -459,8 +459,8 @@ phsource/phonemes.stamp: \
 	phsource/ph_yue \
 	phsource/phonemes \
 	src/espeak-ng
-	ESPEAK_DATA_PATH=$(CURDIR) src/espeak-ng --compile-intonations && \
-		ESPEAK_DATA_PATH=$(CURDIR) src/espeak-ng --compile-phonemes && \
+	ESPEAK_DATA_PATH=$(CURDIR) $(HOST_BUILD_DIR)/src/espeak-ng --compile-intonations && \
+		ESPEAK_DATA_PATH=$(CURDIR) $(HOST_BUILD_DIR)/src/espeak-ng --compile-phonemes && \
 		touch $@
 
 ##### android targets:
@@ -519,7 +519,7 @@ dictsource/%_extra:
 espeak-ng-data/%_dict: src/espeak-ng phsource/phonemes.stamp
 	@echo "  DICT      $@"
 	rm -f $@
-	cd dictsource && ESPEAK_DATA_PATH=$(CURDIR) ../src/espeak-ng --compile="$(*F)"
+	cd dictsource && ESPEAK_DATA_PATH=$(CURDIR) $(HOST_BUILD_DIR)/src/espeak-ng --compile="$(*F)"
 
 dictionaries: \
 	espeak-ng-data/af_dict \
@@ -1000,7 +1000,7 @@ espeak-ng-data/yue_dict: src/espeak-ng p
 endif
 	@echo "  DICT      $@"
 	rm -f $@
-	cd dictsource && ESPEAK_DATA_PATH=$(CURDIR) LD_LIBRARY_PATH=../src:${LD_LIBRARY_PATH} ../src/espeak-ng --compile=yue && cd ..
+	cd dictsource && ESPEAK_DATA_PATH=$(CURDIR) $(HOST_BUILD_DIR)/src/espeak-ng --compile=yue && cd ..
 
 dictsource/yue_emoji:
 	@echo "  EMOJI     $@"
@@ -1065,4 +1065,4 @@ mbrola: \
 
 espeak-ng-data/mbrola_ph/%_phtrans: phsource/mbrola/% src/espeak-ng
 	mkdir -p espeak-ng-data/mbrola_ph
-	ESPEAK_DATA_PATH=$(CURDIR) src/espeak-ng --compile-mbrola=phsource/mbrola/$*
+	ESPEAK_DATA_PATH=$(CURDIR) $(HOST_BUILD_DIR)/src/espeak-ng --compile-mbrola=phsource/mbrola/$*
