--- a/conf/msd_minimal.conf
+++ b/conf/msd_minimal.conf
@@ -126,8 +126,8 @@ available.
 
 
 	<channelList>
-		<includeFile>/root/msd/msd_channels.conf</includeFile>
-		<includeFile>-/root/msd/msd_channels_lanport.conf</includeFile>
+		<includeFile>/opt/etc/msd/msd_channels.conf</includeFile>
+		<includeFile>/opt/etc/msd/msd_channels_lanport.conf</includeFile>
 		<c-hannel>
 			<name>Test1.ts</name> <!-- Channel name, used in request: /channel/... . -->
 			<hubProfileName>default</hubProfileName> <!-- Profile template. -->
