--- a/libfuse/Makefile
+++ b/libfuse/Makefile
@@ -130,8 +130,8 @@ clean:
 distclean: clean
 
 strip:
-	strip --strip-all build/mount.mergerfs
-	strip --strip-all build/mergerfs-fusermount
+	$(STRIP) --strip-all build/mount.mergerfs
+	$(STRIP) --strip-all build/mergerfs-fusermount
 
 install-utils: mergerfs-fusermount mount.mergerfs strip
 	install -D build/mergerfs-fusermount "$(INSTALLBINDIR)/mergerfs-fusermount"
