--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -139,7 +139,7 @@ if(CMAKE_SYSTEM_NAME STREQUAL "FreeBSD")
         message(FATAL_ERROR "/usr/local/include/gnu/regex.h missing, did you install libgnuregex?")
     endif()
 else ()
-    add_definitions(-DCONFIG_FILE=\"/etc/multitail.conf\")
+    add_definitions(-DCONFIG_FILE=\"/opt/etc/multitail.conf\")
 endif()
 
 # Set ``CURSES_NEED_NCURSES`` to ``TRUE`` before the
